hibernate-core/src/main/java/org/hibernate/internal/util/collections/BoundedConcurrentHashMap.java

```diff
@@ -225,6 +225,9 @@ public int hashCode() {
 
 		@Override
 		public boolean equals(Object o) {
+			if (!(o instanceof HashEntry)) {
```

Comment:
`HashEntry` is a private class and we ensured this method is never called with an argument of a different type, so this check is unnecessary.